Makefile

```diff
@@ -27,7 +27,7 @@ include ../phoenix-rtos-build/Makefile.common
 include ../phoenix-rtos-build/Makefile.$(TARGET_SUFF)
 
 CFLAGS += $(BOARD_CONFIG)
-CFLAGS += -I. -DHAL=\"hal/$(TARGET_SUFF)/hal.h\" -DVERSION=\"$(VERSION)\"
+CFLAGS += -I. -DHAL=\"hal/hal.h\" -DVERSION=\"$(VERSION)\"
```

Comment:
But there is an error:
```
<command-line>: fatal error: hal/hal.h: No such file or directory
```

Reply:
As I wrote in a description, this PR is a part of the bigger changes in the kernel. To make review easier, the changes have been dividing into several parts, after reviewing all of them, the code will compile.
This PR will be merge to temporary branch `feature/syspage` so that it will not corrupt current master.